src/main/scala/game/Game.scala

```diff
@@ -1,23 +1,52 @@
 import scala.util.Random
 import command.Command
-import command.PlayAgainCommand
-import command.ExitCommand
-import command.Play._
-import command.Play
+import command.Command._
+import command.Move
+import command.Move._
+import error.Error
+import error.InvalidInput
+import error.Printable
 
 object Game extends App {
 
-  def provideCPUMove(): PlayCommand = Random.shuffle(Play.moves).head
+  implicit val printError = new Printable[Error] {
+    def providePrintableMsg(err: Error): String = s"--> [${err.message}] !!!"
+  }
+
+  def printMessage[T](e: T)(implicit d: Printable[T]): Unit = println(d.providePrintableMsg(e))
+
+  def isExitCommandDetect(value: String): Boolean = value == "Q"
+
+  def provideCPUMove(): Move = Random.shuffle(Move.moves).head
 
-  def parsePlayerMove(value: String): Either[Error, Command] = value match {
-    case "0" => Right(Rock);
-    case "1" => Right(Paper);
-    case "2" => Right(Scissor);
-    case "Q" => Right(ExitCommand)
-    case _ => Left(InvaliInput)
+  def parsePlayerMove(value: String): Either[Error, Move] = value match {
+    case "0" => Right(Rock)
+    case "1" => Right(Paper)
+    case "2" => Right(Scissor)
+    case _ => Left(InvalidInput)
   }
 
-  def evaluateWinner(userPlay: PlayCommand, cpuPlay: PlayCommand): String =
+  def handleInvalidInput(error: Error) = {
+    printMessage(error);
+    PlayAgain
+  }
+
+  def handleUserMove(userMove: Either[Error, Move]): Command.PlayAgain.type =
+    userMove.fold(
+      handleInvalidInput,
+      (userMove: Move) => {
+
+        val cpuMove = provideCPUMove()
+        println(s"[User] ${userMove.toString()} <-> ${cpuMove.toString()} [CPU]")
```

Comment:
ci sta farlo esplicito, ma sappi che `.toString` viene invocato in automatico quando fai string interpolation